src/test/util/setup_common.cpp

```diff
@@ -103,10 +103,12 @@ TestingSetup::TestingSetup(const std::string& chainName) : BasicTestingSetup(cha
     g_rpc_node = &m_node;
     RegisterAllCoreRPCCommands(tableRPC);
 
+    g_rpc_node->scheduler = MakeUnique<CScheduler>();
```

Comment:
style-nit in commit 047c1fd23ab7f281af7acd238f3bbb441817686c:

`g_rpc_node` might go away (#17548), and using `m_node` directly would be cleaner code anyway.

Reply:
Also, the pointer should probably be reset in `~TestingSetup`

Reply:
yeah totally. do you think the lack of reset could have somehow caused the failure [here](https://travis-ci.org/bitcoin/bitcoin/jobs/649620800)? I'll push shortly and observe, but still trying to fill in my conceptual understanding.

not directly related, but another thing I'm wondering is why the scheduler thread isn't interrupted [here](https://github.com/bitcoin/bitcoin/blob/master/src/init.cpp#L162)